vspace/src/main/java/edu/asu/diging/vspace/core/references/impl/CoinSMetadataProvider.java

```diff
@@ -0,0 +1,246 @@
+        return ReferenceMetadataType.DEFAULT;
+    }
+
+    private String getReferenceData(String referenceType) {
+        return referenceMap.getOrDefault(referenceType, CoinSConstants.DEFAULT_RFT_VAL_FMT);
+    }
+
+    @Override
+    public StringBuffer getReferenceMetadata(Reference reference) throws ReferenceMetadataEncodingException {
+
+        String refTitleEncoded = getRefTitleEncoded(reference.getTitle(), reference.getType());
+        String refAuthorEncoded = getRefAuthorEncoded(reference.getAuthor()) + getRefYearEncoded(reference.getYear());
+        String refJournalEncoded = getRefJournalEncoded(reference.getJournal()) + getRefUrlEncoded(reference.getUrl());
+        String refVolumeEncoded = getRefVolumeEncoded(reference.getVolume()) + getRefIssueEncoded(reference.getIssue());
+        String refPagesEncoded = getRefPagesEncoded(reference.getPages())
+                + getRefEditorsEncoded(reference.getEditors());
+        String refNoteEncoded = getRefNoteEncoded(reference.getNote()) + getRefTypeEncoded(reference.getType());
+        String referenceTypeRFTValueString = getReferenceData(reference.getType());
+        StringBuffer urlEncodedReferenceMetaData = new StringBuffer(CoinSConstants.DEFAULT_URL_VERSION);
+        try {
+            urlEncodedReferenceMetaData.append("&");
+            urlEncodedReferenceMetaData.append(CoinSConstants.DEFAULT_CTX_VERSION);
+            urlEncodedReferenceMetaData.append(CoinSConstants.RFR_ID_TAG);
+            urlEncodedReferenceMetaData
+                    .append(URLEncoder.encode(CoinSConstants.DEFAULT_RFR_ID, StandardCharsets.UTF_8.name()));
+            urlEncodedReferenceMetaData.append(CoinSConstants.RFT_VAL_FMT_TAG);
+            urlEncodedReferenceMetaData.append(referenceTypeRFTValueString);
+            urlEncodedReferenceMetaData.append(refTitleEncoded);
+            urlEncodedReferenceMetaData.append(refAuthorEncoded);
+            urlEncodedReferenceMetaData.append(refJournalEncoded);
+            urlEncodedReferenceMetaData.append(refVolumeEncoded);
+            urlEncodedReferenceMetaData.append(refPagesEncoded);
+            urlEncodedReferenceMetaData.append(refNoteEncoded);
+
+        } catch (UnsupportedEncodingException e) {
+            throw new ReferenceMetadataEncodingException(e);
+        }
+        urlEncodedReferenceMetaData.append(CoinSConstants.RFT_DEFAULT_LANGUAGE);
+        return urlEncodedReferenceMetaData;
+    }
+
+    private String getRefTypeEncoded(String type) throws ReferenceMetadataEncodingException {
+        if (type == null || type.trim().equals("")) {
+            return "";
+        }
+        try {
+            if (type.equals(CoinSConstants.BLOGPOST_TYPE) || type.equals(CoinSConstants.PODCAST)
+                    || type.equals(CoinSConstants.PRESENTATION) || type.equals(CoinSConstants.SOFTWARE_DOCUMENTATION)) {
+                return CoinSConstants.RFT_TYPE_TAG + URLEncoder.encode(type, StandardCharsets.UTF_8.name());
```

Comment:
THere are a couple of things I would change for all the return statements like this:
- the constant should not include `&` and `=`. It should just have the attribute name. The way it is right now, we could not reuse it somewhere else where no `=` or `&` is needed.
- create a method that does this and catches/throws the exception. Then you can get rid of all the try/catch blocks and just keep the `throws` part in the method signature. Also, right now there are different approaches used (`+` and `format()`), so that would make that consistent. And if we need to change the charset, it's just one line not several that needs to be changed.